Tests/Marketplace/search_and_install_packs.py

```diff
@@ -413,381 +600,348 @@ def install_packs(client: demisto_client,
 
-    # make the pack installation request
-    try:
-        response_data, status_code, _ = client.api_client.call_api(resource_path='/contentpacks/installed/upload',
-                                                                   method='POST',
-                                                                   auth_settings=auth_settings,
-                                                                   header_params=header_params, files=files,
-                                                                   response_type='object')
-
-        if 200 <= status_code < 300:
-            logging.info(f'All packs from file {pack_path} were successfully installed on server {host}')
-        else:
-            message = response_data.get('message', '')
-            raise Exception(f'Failed to install packs - with status code {status_code}\n{message}')
-    except Exception:
-        logging.exception('The request to install packs has failed.')
-        sys.exit(1)
+    api_data = get_pack_dependencies(client=client, pack_id=pack_id)
 
+    if not api_data:
+        return pack_id, []
 
-def search_and_install_packs_and_their_dependencies_private(test_pack_path: str,
-                                                            pack_ids: list,
-                                                            client: demisto_client):
-    """ Searches for the packs from the specified list, searches their dependencies, and then installs them.
-    Args:
-        test_pack_path (str): Path of where the test packs are located.
-        pack_ids (list): A list of the pack ids to search and install.
-        client (demisto_client): The client to connect to.
-
-    Returns (list, bool):
-        A list of the installed packs' ids, or an empty list if is_nightly == True.
-        A flag that indicates if the operation succeeded or not.
-    """
-    host = client.api_client.configuration.host
-
-    logging.info(f'Starting to search and install packs in server: {host}')
-
-    install_packs_private(client, host, pack_ids, test_pack_path)
+    dependencies_data: list[dict] = create_dependencies_data_structure(
+        response_data=api_data.get("dependencies", []),
+        dependants_ids=[pack_id],
+        checked_packs=[pack_id],
+    )
+    pack_api_data = api_data["packs"][0]
+    current_packs_to_install = [pack_api_data]
 
-    return SUCCESS_FLAG
+    if dependencies_data:
+        logging.debug(
+            f"Found dependencies for '{pack_id}': {[dependency['id'] for dependency in dependencies_data]}"
+        )
 
+        for dependency in dependencies_data:
+            if is_pack_deprecated(
+                pack_id=dependency["id"],
+                production_bucket=production_bucket,
+                pack_api_data=dependency,
+            ):
+                logging.critical(
+                    f"Pack '{pack_id}' depends on pack '{dependency['id']}' which is a deprecated pack."
+                )
+                return pack_id, None
```

Comment:
Line 800 is like this:
```suggestion
                return pack_id, []
```

Reply:
same as above: this is the desired behaviour, either None for Error, or a list, that could be empty.